pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -853,6 +850,13 @@ monitorLoop:
 			}
 
 		case libvirt.DOMAIN_JOB_NONE:
+			// check if an ongoing migration has completed before we could capture the outcome
+			// in that case, get the last migration result and set the status according to it.
+			if lastProgressUpdate != start {
+				logger.Info("Migration job has probably completed before we could capture the status. Getting latest status.")
+				activeDomainJobStatsFlag = libvirt.DOMAIN_JOB_STATS_KEEP_COMPLETED
```

Comment:
It may make sense to split this a little. Can we structure this more like this:

```go
 if runningJobExists {
      interpretJob(job)
 } else if completedJobExists {
     interpretJob(job)
 } else {
   // can this happen?
 }
```

Let's untangle the query change and the job interpretation.